hydra-task/src/main/java/com/addthis/hydra/task/output/tree/ReadOnceListLeakDetection.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.addthis.hydra.task.output.tree;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+import io.netty.util.ResourceLeak;
+
+/**
+ * Collection who contents can either be read exactly once or
+ * whose contents can be released
+ * @param <T>
+ */
+@NotThreadSafe
+public class ReadOnceListLeakDetection<T> implements ReadOnceList<T> {
+
+    @Nonnull
+    protected ResourceLeak leak;
```

Comment:
can these be final? I can see they are protected, so maybe you have a use case in mind where they change, but having trouble finding it